### PR TITLE
openfoam: add versions 2306, 2212_230612 (patch), 2212

### PR DESCRIPTION
**Pull request by olesenm**
Add in checksum information for latest and recently patched 2212 version.

- supersedes https://github.com/spack/spack/pull/38093/

### CODE REVIEW

**Comment by olesenm** [comment]
Anyone? @alalazo , @adamjstewart

**Review by adamjstewart — DISMISSED** [review]
Sorry for the delay! Feel free to annoy people to review if more than a week has gone by. There is also a #pull-requests channel on Slack where you can ask for reviews if no one is responding on GitHub.

File: var/spack/repos/builtin/packages/openfoam/package.py [hunk 1]
```diff
@@ -267,6 +267,12 @@ class Openfoam(Package):
 
     version("develop", branch="develop", submodules="True")
     version("master", branch="master", submodules="True")
+    version("2306", sha256="d7fba773658c0f06ad17f90199565f32e9bf502b7bb03077503642064e1f5344")
+    version
+    (
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openfoam/package.py]:
```suggestion
    version(
```

**Review by adamjstewart — APPROVED** [review]

**Comment by olesenm** [comment]
Thanks!

**Comment by olesenm** [comment]
> Sorry for the delay! Feel free to annoy people to review if more than a week has gone by. There is also a #pull-requests channel on Slack where you can ask for reviews if no one is responding on GitHub.

No worries - I was doing a bunch of other things before getting back to check the status.

**Pull request merged**